.github/actions/check_version/get_version_from_files.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+import os, sys
+import argparse
+import struct
+
+
+#
+# MARK: - argparse
+#
+
+
+parser = argparse.ArgumentParser(
+    description="Check os version compliance with standards used in the project"
+)
+
+parser.add_argument("firmware_hex_path", type=str, help="path of firmware hex file")
+parser.add_argument("os_bin_path", type=str, help="path of os binary file")
+
+args = parser.parse_args()
+
+
+#
+# MARK: - Functions
+#
+
+
+def addToEnvironement(data: str):
+    env_file = os.getenv("GITHUB_ENV")
+    with open(env_file, "a") as env:
+        env.write(data + "\n")
+
+
+def getDataFromFile(file_path: str, mode="r") -> bytes:
+    data: bytes
+    try:
+        f = open(file_path, mode)
+        data = f.read()
+        f.close()
+    except BaseException as error:
+        print(f"{error}")
+        sys.exit(1)
+
+    return data
+
+
+def getOSVersion():
+    os_version_raw = getDataFromFile(args.os_bin_path, mode="rb")[20:28]
+
+    os_version_major = os_version_raw[0]
+    os_version_minor = os_version_raw[1]
+    os_version_revision = struct.unpack("h", os_version_raw[2:4])[0]
+    os_version_build = struct.unpack("I", os_version_raw[4:8])[0]
+
+    addToEnvironement(f"OS_VERSION_FILE_MAJOR={os_version_major}")
+    addToEnvironement(f"OS_VERSION_FILE_MINOR={os_version_minor}")
+    addToEnvironement(f"OS_VERSION_FILE_REVISION={os_version_revision}")
+    addToEnvironement(f"OS_VERSION_FILE_BUILD={os_version_build}")
+
+
+def getFirmwareVersion():
+    firmware_raw = getDataFromFile(args.firmware_hex_path)
+
+    START_OS_MARKER = ":020000040804EE\n"
+    START_VERSION_MARKER = 18
+    VERSION_SIZE = 16
+
+    pos_os_start = firmware_raw.find(START_OS_MARKER)
+    pos_version_line = firmware_raw.find("\n", pos_os_start + len(START_OS_MARKER) + 1)
+    pos_version_start_line = firmware_raw.find("\n", pos_version_line)
+
+    firmware_version_raw = firmware_raw[
+        pos_version_start_line
+        + START_VERSION_MARKER : pos_version_start_line
+        + START_VERSION_MARKER
+        + VERSION_SIZE
+    ]
+
+    firmware_version_major = int(firmware_version_raw[0:2], 16)
+    firmware_version_minor = int(firmware_version_raw[2:4], 16)
+    firmware_version_revision = int(
+        firmware_version_raw[6:8] + firmware_version_raw[4:6], 16
+    )
+    firmware_version_build = int(
+        firmware_version_raw[14:16]
+        + firmware_version_raw[12:14]
+        + firmware_version_raw[10:12]
+        + firmware_version_raw[8:10],
+        16,
+    )
+
+    addToEnvironement(f"FIRMWARE_VERSION_FILE_MAJOR={firmware_version_major}")
+    addToEnvironement(f"FIRMWARE_VERSION_FILE_MINOR={firmware_version_minor}")
+    addToEnvironement(f"FIRMWARE_VERSION_FILE_REVISION={firmware_version_revision}")
+    addToEnvironement(f"FIRMWARE_VERSION_FILE_BUILD={firmware_version_build}")
```

Comment:
why not export the whole string? do you need the independent parts?
you can do both, so that you can use the whole version when needed and the individual parts where needed as well